### PR TITLE
using mdgen instead of lean2md

### LINKED ISSUE #122: using mdgen instead of lean2md
generate md files in pure lean

* mdgen https://github.com/Seasawher/mdgen

* see https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/mdgen.3A.20lean2md.20in.20pure.20lean

### PR DESCRIPTION
**Pull request by Seasawher**
resolve  #122

@arthurpaulino Replacing lean2md with mdgen does not seem to break the markdown file. However, the contents of the files in the lean directory are different from the contents of the files in the md directory, resulting in a large number of diffs. What should I do?

Should the contents of the `lean` directory be respected or the `md` directory?

### CODE REVIEW

**Comment by Seasawher** [comment]
By the way, since the contents of the `md` directory can be generated from the `lean` directory, there is no need to version control the `md` directory.

File: lakefile.lean [hunk 1]
```diff
@@ -17,23 +20,3 @@ def runCmd (cmd : String) (args : Array String) : ScriptM Bool := do
   if hasError then
     IO.eprint out.stderr
   return hasError
-
-script build do
```
Comment (Julian) [inline lakefile.lean:21]: Perhaps it's worth leaving `lake run build` in place and simply updating it -- that makes it slightly easier for someone to build the `md` files if needed, without needing to look up `mdgen`'s CLI syntax.

**Review by Julian — COMMENTED** [review]
> Should the contents of the lean directory be respected or the md directory?

The diff here looks correct (i.e. the changes look like an improvement) -- probably what happened was that  lean2md wasn't run on the `.lean` files -- you generated the PR from those (the `.lean` files), correct?

**Comment by Julian** [comment]
Thanks! This looks pretty good to me! I didn't try running mdgen yet, but I'll do that.

As for

> By the way, since the contents of the md directory can be generated from the lean directory, there is no need to version control the md directory. If you need a human-readable format, you can host it on the web using mdbook. see https://github.com/leanprover-community/lean4-metaprogramming-book/pull/121

That sounds good to me as well, will have a look at that PR.

**Comment by Seasawher** [comment]
@Julian I have made corrections to the points you pointed out.

**Comment by Julian** [comment]
This looks good I think! Let's merge and see how it goes. Thanks!

**Pull request merged**